daemon/mqtt/matrix-mqtt

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import codecs
+import paho.mqtt.client as mqtt
+from os import kill, system
+import signal
+import subprocess
+
+MQTT_HOSTNAME = '172.23.4.6'
+MQTT_USERNAME = ''
+MQTT_PASSWORD = ''
+TOPIC_BASE = 'Kitchen/Matrix/'
+TOPIC_ANIM = TOPIC_BASE + 'Animation'
+TOPIC_LWT = TOPIC_BASE + 'lwt'
+TOPIC_REBOOT = TOPIC_BASE + 'reboot'
+TOPIC_SHUTDOWN = TOPIC_BASE + 'shutdown'
+
+# no root no fun.
+ANIMATION_DICT = {
+    1: 'matrix-clock',
+    2: 'matrix-analogclock',
+    3: 'matrix-chess',
+    4: 'matrix-fullblink',
+    5: 'matrix-cairoexample',
+    6: 'matrix-circle',
+    7: 'matrix-fire',
+    8: 'matrix-zwickl',
+    9: 'matrix-rain'
+}
+
+cur_anim_pid = -1
+
+def on_message_animation(client, userdata, message):
+    global cur_anim_pid
+
+    no = int(message.payload.decode("utf-8"))
+
+    if no >= 100:
+        if no == 102:
+            if cur_anim_pid != -1:
+                kill(cur_anim_pid, signal.SIGKILL)
+            cur_anim_pid = -1
+        else:
+            print('Go home kiddy!')
+
+        print('Aye, aye!')
+
+    if not no in ANIMATION_DICT:
+        print('Go home kiddy!')
+    
+    if cur_anim_pid != -1:
+        kill(cur_anim_pid, signal.SIGKILL)
+    
+    proc = subprocess.Popen([ANIMATION_DICT.get(no), '-s', 'localhost'])
+    cur_anim_pid = proc.pid
+    print('Aye, aye, captain!')
+
+def on_message_reboot(client, userdata, message):
+    if not message.retain:
+        system("reboot")
+
+def on_message_shutdown(client, userdata, message):
+    if not message.retain:
+        system("poweroff")
```

Comment:
Can't we use systemd-poweroff service to power off the system and grant the user poweroff abilities? This would avoid running the script as root.

Reply:
Didn't know that this is even possible. Will search for some documentation

Reply:
If you run a KDE session and press the shutdown button, then your user also must somehow have the permission to shutdown the system.

Seems liks this is done via dbus or console-kit.

That's worth a look:
https://stackoverflow.com/questions/23013274/shutting-down-computer-linux-using-python

Maybe we can instrument python to perform the shutdown directly.

Reply:
Not sure if this is a good idea, though.

Reply:
That looks a bit easier: https://wiki.ubuntuusers.de/systemd/systemctl/#System-Kommandos
Just call `systemctl reboot`

Reply:
Great! We only need to make sure that a user may invoke this command.